src/context/WalletProvider/NativeWallet/components/LegacyLogin.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  Button,
+  Checkbox,
+  FormControl,
+  FormErrorMessage,
+  Input,
+  ModalBody,
+  ModalHeader,
+  useColorModeValue,
+} from '@chakra-ui/react'
+import { useState } from 'react'
+import { FieldValues, useForm } from 'react-hook-form'
+import { useTranslate } from 'react-polyglot'
+import { RouteComponentProps } from 'react-router-dom'
+import { Card } from 'components/Card/Card'
+import { Text } from 'components/Text'
+
+// @TODO(NeOMakinG): Remove this and add the 2fa code logic
+const twoFactorAuthCode = ''
+
+export const LegacyLogin = ({ history }: RouteComponentProps) => {
+  const [isCaptchaValidated, setCaptchaValidated] = useState(false)
+  const captchaBgColor = useColorModeValue('gray.50', 'gray.700')
+  const checkboxBorderColor = useColorModeValue('gray.400', 'white')
+
+  const {
+    handleSubmit,
+    register,
+    reset,
+    formState: { errors, isSubmitting },
+  } = useForm({ shouldUnregister: true })
+
+  const translate = useTranslate()
+
+  const onSubmit = async (values: FieldValues) => {
+    // Reset every fields in order to remove the password and email from the memory
+    reset()
+    history.push('/native/legacy/two-factor')
```

Comment:
I'd do something like this to prepare for the async call.

Also, we can't remove the password yet, since if there is 2FA required, we'll need the password later on.
```suggestion
try {
    await new Promise((resolve, reject) => resolve())
    // TODO: use response to create wallet vault.
    history.push('/native/create', { vault })
} catch (err) {
    history.push('/native/legacy/two-factor')
}
```